### ISSUE TITLE
Support dPoP   (OAuth 2.0 Demonstrating Proof-of-Possession at the Application Layer)

### ISSUE THREAD

**Issue by damooo** [ory/hydra#3235]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/hydra/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/hydra/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [ ] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe your problem

> [DPoP](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-dpop) (for Demonstrating Proof-of-Possession at the Application Layer)
   is an application-level mechanism for sender-constraining OAuth
   access and refresh tokens.  It enables a client to prove the
   possession of a public/private key pair by including a DPoP header in
   an HTTP request.

New w3c standards like `solid-oidc` mandates dPoP, to enable decentralized and secure identities. It will be great if hydra supports it.

### Describe your ideal solution

Supporting dpop

### Workarounds or alternatives

Is it possible to layer over hydra for it? Any little guidance will be appreciated. 

### Version

Current

**Comment by froger** [comment]
Yes to this! Hydra and its admin have almost everything to be a headless solid-oidc provider. (referring to the [rfc](https://solid.github.io/solid-oidc/primer/), and an [implementation](https://github.com/solid-contrib/solid-oidc-provider)).

The missing parts are these two steps:

> 16. Validates DPoP Token Signature
> The OP extracts the client’s public key from the DPoP header (at header.jwk). It confirms that the DPoP token has a valid signature. If not, the OP must reject the request with a 400 HTTP status and invalid_dpop_proof error code.

> 17. Converts the DPoP public key to a JWK thumbprint
> Currently the DPoP token contains a JWK public key, but before we place it inside the access token, it needs to be converted into a [JWK thumbprint](https://tools.ietf.org/html/rfc7638). Our JWK thumbprint looks more like:
> `9XmwK8mQ3H5-PnzAt3lFHzWBW_v5QhYynezbbit4kC8`

So adding [dPoP support](https://datatracker.ietf.org/doc/html/draft-fett-oauth-dpop-02) would be greatly appreciate, I can not really help on the implementation… So hope these references are helpful to do triage. 

Thanks

**Comment by kmherrmann** [comment]
DPoP support would indeed be a great addition. A PR would be highly appreciated! :)

**Comment by damooo** [comment]
DPoP is now an rfc.

**Comment by rektide** [comment]
Maybe worth mentioning that hydra's underlying fosite library gained dpop support in fosite#641.

**Comment by rektide** [comment]
Maybe worth mentioning that hydra's underlying fosite library gained dpop support in ory/fosite#641.

**Comment by rektide** [comment]
Worth noting that there was a ory/fosite#641, closed as stale.
